src/vdom/diff.js

```diff
@@ -73,8 +73,14 @@ function idiff(dom, vnode, context, mountAll) {
 	}
 
 
-	// empty values (null & undefined) render as empty Text nodes
-	if (vnode==null) vnode = '';
+	if (vnode==null) {
+		if (componentRoot) {
+			if (dom) recollectNodeTree(dom);
+			return document.createComment('');
```

Comment:
I think we might have to check if `dom` is already a Comment node here and retain it, otherwise the Component will get recycled.